plugins/pulp_python/plugins/models.py

```diff
@@ -1,120 +1,85 @@
-from gettext import gettext as _
 import hashlib
-import re
-import tarfile
 
 from mongoengine import StringField
 from pulp.server.db.model import FileContentUnit
+import twine  # noqa
 
 from pulp_python.common import constants
 
 
 DEFAULT_CHECKSUM_TYPE = 'sha512'
-
-
-# These are required to be in the PKG-INFO file.
-REQUIRED_ATTRS = ('name', 'version', 'summary', 'home_page', 'author', 'author_email', 'license',
-                  'description', 'platform')
+REQUIRED_ATTRS = ('name', 'version', 'author', 'author_email', 'description', 'summary')
 
 
 class Package(FileContentUnit):
     """
     This class represents a Python package.
     """
 
-    # unit key
     name = StringField(required=True)
     version = StringField(required=True)
-
-    author = StringField()
+    filename = StringField(required=True)
+    author = StringField(required=True)
     author_email = StringField()
     description = StringField()
-    home_page = StringField()
-    license = StringField()
-    platform = StringField()
     summary = StringField()
 
     _checksum = StringField()
     _checksum_type = StringField(default=DEFAULT_CHECKSUM_TYPE)
-    _filename = StringField()
-    _metadata_file = StringField()
 
     # For backward compatibility
     _ns = StringField(default='units_python_package')
     _content_type_id = StringField(required=True, default=constants.PACKAGE_TYPE_ID)
 
-    unit_key_fields = ('name', 'version')
+    unit_key_fields = ('filename',)
 
     meta = {
         'allow_inheritance': False,
         'collection': 'units_python_package',
-        'indexes': [],
+        'indexes': [{'fields': ['-filename'], 'unique': True}],
     }
 
     @classmethod
-    def from_archive(cls, archive_path):
+    def from_json(cls, file_data, release, distribution_data):
         """
-        Instantiate a Package using the metadata found inside the Python package found at
-        archive_path. This tarball should be the build result of running setup.py sdist on the
-        package, and should contain a PKG-INFO file. This method will read the PKG-INFO to determine
-        the package's metadata and unit key.
-
-        :param archive_path: A filesystem path to the Python source distribution that this Package
-                             will represent.
-        :type  archive_path: basestring
-        :return:             An instance of Package that represents the package found at
-                             archive_path.
-        :rtype:              pulp_python.plugins.models.Package
-        :raises:             ValueError if archive_path does not point to a valid Python tarball
-                             created with setup.py sdist.
-        :raises:             IOError if the archive_path does not exist.
+        Create and return (but do not save) an instance of a single python package object from the
+        metadata available from PyPI JSON.
+
+        :param file_data: metadata specific to a file
```

Comment:
I'm not exactly sure what this means. Which file? At first I thought it was from the JSON from the PyPI API, but I see there's a second dictionary, distribution_data. How do they differ and where do they come from?